docs/root/configuration/listeners/udp_filters/dns_filter.rst

```diff
@@ -54,8 +63,71 @@ Example Configuration
               endpoint:
                 address_list:
                   address:
-                    - 10.0.3.1
+                  - 10.0.3.1
 
 
 In this example, Envoy is configured to respond to client queries for three domains. For any
-other query, it will forward upstream to external resolvers.
+other query, it will forward upstream to external resolvers. The filter will return an address
+matching the input query type. If the query is for type A records and no A records are configured,
+Envoy will return no addresses and set the response code appropriately. Conversely, if there are
+matching records for the query type, each configured address is returned. This is also true for
+AAAA records. Only A and AAAA records are supported. If the filter parses other queries for other
+record types, the filter immediately responds indicating that the query is not supported.
+
+To disable external resolution, one can omit the `client_config` section of the config. Envoy interprets
+this configuration to mean that name resolution should be done only from the data appearing in the
+`server_config` section. A query for a name not appearing in the DNS table will receive a "No Answer"
+DNS response.
+
+The filter can also consume its configuration from an external dns table. The same configuration
+that appears in the static configuration can be stored in a Proto3-conformant JSON file and
+referenced in the configuration using the :ref:`external_dns_table DataSource <envoy_api_msg_core.DataSource>`
+directive:
+
+Example External DnsTable Configuration
+---------------------------------------
+
+.. code-block:: yaml
+
+    listener_filters:
+      name: "envoy.filters.udp.dns_filter"
+      typed_config:
+        '@type': 'type.googleapis.com/envoy.extensions.filters.udp.dns_filter.v3alpha.DnsFilterConfig'
+        stat_prefix: "my_prefix"
+        server_config:
+          external_dns_table:
+            filename: "/home/ubuntu/configs/dns_table.json"
+
+In the file, the table can be defined as follows:
+
+DnsTable JSON Configuration
+---------------------------
+
+.. code-block:: text
```

Comment:
nit: I'm guessing sphinx supports json formatting. Would it be better to use YAML though? Up to you.

Reply:
With `json` formatting I have to have quotes around the key names which does not accurately reflect the DataSource file.   I think I tried `yaml`.  It doesn't stand out because the docs build still reported warnings.  `text` was the only value that allowed the code block in the document.

Reply:
In the event I misunderstood your comment, I did try using yaml as the external data source, but could not get that to work.  I proceeded using the Proto3 json format that's documented in the RST.

Reply:
Both YAML and JSON should be covertable to Proto (see the message utility functions that we use). I would suggest supporting both, mainly because YAML is easier to read and comment and when it's read into a schema it's pretty safe, but up to you. Feel free to handle in a follow up.

Reply:
I will handle this in a follow up PR since the code for this doesn't appear in this PR.